kernel/module.c

```diff
@@ -1827,6 +1828,7 @@ static void unset_module_init_ro_nx(struct module *mod) { }
 void __weak module_free(struct module *mod, void *module_region)
 {
 	vfree(module_region);
+	kasan_module_free(module_region);
```

Comment:
Sorry, but I didn't get this. This seems like a right place to me for arches that use default module_free() (x86).

 How from 
> module_free is __weak and intended as default implementation unless arch wants to override it

follows that

> This is a wrong place to insert the call.

Please elaborate.